source/common/http/path_utility.cc

```diff
@@ -69,5 +68,15 @@ void PathUtil::mergeSlashes(HeaderMap& headers) {
       prefix, absl::StrJoin(absl::StrSplit(path, '/', absl::SkipEmpty()), "/"), query, suffix));
 }
 
+absl::string_view PathUtil::removeQueryAndFragment(const absl::string_view path) {
+  absl::string_view ret = path;
+  // Trim query parameters and/or fragment if present.
+  size_t offset = ret.find_first_of("?#");
```

Comment:
Note: `#` should be never sent on the wire (at least in theory), so we might want to kill such connections... but this would require investigation to make sure that this wouldn't kill some popular but broken clients.

Could you open an issue to follow-up on this?

Reply:
That's a good point, I created the issue here:  https://github.com/envoyproxy/envoy/issues/10033